From 111148d470e13dcc2cd60ba04e8d6d473a374b10 Mon Sep 17 00:00:00 2001
From: Daniel Molkentin <dmolkentin@suse.com>
Date: Thu, 26 Mar 2020 16:48:56 +0100
Subject: [PATCH] Correctly link with --as-needed

Linker needs to analzye the objects first before considering libraries.
as-needed is commonly used on distros.
---
 modules/60crypt-ssh/helper/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/60crypt-ssh/helper/Makefile b/modules/60crypt-ssh/helper/Makefile
index 7abc5f3..9aa5003 100644
--- a/modules/60crypt-ssh/helper/Makefile
+++ b/modules/60crypt-ssh/helper/Makefile
@@ -20,7 +20,7 @@ console_auth:	auth.c
 	$(CC) $(CFLAGS) $^ -o $@
 
 unlock:	crypttab.o unlock.o
-	$(CC) $(CFLAGS) -lblkid $^ -o $@
+	$(CC) $(CFLAGS) $^ -lblkid -o $@
 
 crypttab-test: crypttab-test.c crypttab.o crypttab-test-data
 	$(CC) crypttab-test.c $(CFLAGS) crypttab.o -lblkid -o crypttab-test
